common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -192,7 +194,18 @@ public TransportClient createClient(String remoteHost, int remotePort)
           logger.info("Found inactive connection to {}, creating a new one.", resolvedAddress);
         }
       }
-      clientPool.clients[clientIndex] = createClient(resolvedAddress);
+      if (System.currentTimeMillis() - clientPool.lastConnectionFailed[clientIndex]
+        < conf.ioRetryWaitTimeMs()) {
+        throw new IOException(
+          String.format("Connecting to %s failed in the last %s ms, fail this connection directly",
+            resolvedAddress, conf.ioRetryWaitTimeMs()));
+      }
+      try {
+        clientPool.clients[clientIndex] = createClient(resolvedAddress);
```

Comment:
Shall we reset waiting time if connection succeeded?

Reply:
thanks, I will set `clientPool.lastConnectionFailed[clientIndex] = 0;` after create client successfully.

Reply:
I'm confused by how it would ever succeed after it failed once because if it failed once then it never gets here to try to create client?  Maybe there is something I'm missing with the clientIndex, but that code above seems like a bug.

Reply:
never mind, you already have time if < conf.ioRetryWaitTimeMs